### PR TITLE
Collect account details for CIS AWS events

### PR DESCRIPTION
**Pull request by ari-aviran**
### Summary of your changes
Collect the following fields for CIS AWS findings:
- `cloud.account.name`
- `cloud.account.id`
- `cloud.provider`

In addition, refactor to the common data provider code to support multiple benchmarks with different types of common data gathered.

### Screenshot/Data
Screenshot 2023-01-29 at 10 54 03

### Related Issues
Resolves https://github.com/elastic/cloudbeat/issues/650
Resolves https://github.com/elastic/cloudbeat/issues/586

### Checklist
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary README/documentation (if appropriate)

### CODE REVIEW

File: dataprovider/k8s_data_provider_test.go [hunk 1]
```diff
@@ -40,30 +39,30 @@ type clusterNameProviderMock struct {
...
-func (c clusterNameProviderMock) GetClusterName(ctx context.Context, cfg *config.Config, log *logp.Logger) (string, error) {
+func (c clusterNameProviderMock) GetClusterName(_ context.Context, _ *config.Config, _ *logp.Logger) (string, error) {
 	if c.clusterName == "error" {
 		return "", os.ErrNotExist
 	}
 	return c.clusterName, nil
 }
 
-func Test_k8sDataCollector_CollectK8sData(t *testing.T) {
+func Test_k8sDataProvider_CollectK8sData(t *testing.T) {
```
Comment (ari-aviran) [inline dataprovider/k8s_data_provider_test.go:49]: Should be called `_GetData`

File: dataprovider/k8s_data_provider.go [hunk 2]
```diff
@@ -56,30 +70,29 @@ func NewK8sDataProvider(log *logp.Logger, cfg *config.Config) k8sDataProvider {
...
 }
 
-func (k k8sDataCollector) CollectK8sData(ctx context.Context) *CommonK8sData {
+func (k k8sDataProvider) FetchData(ctx context.Context) (CommonData, error) {
 	if k.kubeClient == nil {
 		k.log.Debug("Could not collect Kubernetes common data as the client was not provided")
-		return nil
-
+		return nil, nil
 	}
```
Comment (uri-weisman) [inline dataprovider/k8s_data_provider.go:85]: If we return an error in `NewK8sDataProvider` this logic can be removed.

File: resources/providers/awslib/iam/account_alias.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+// under the License.
+
+package iam
+
+import (
+	"context"
+	iamsdk "github.com/aws/aws-sdk-go-v2/service/iam"
+)
+
+func (p Provider) GetAccountAlias(ctx context.Context) (string, error) {
```
Comment (uri-weisman) [inline resources/providers/awslib/iam/account_alias.go:25]: Can u please add a test for that?

**Review by uri-weisman — CHANGES_REQUESTED** [review]
Really loved the new design! minor comments...

File: dataprovider/common_data_provider.go [hunk 4]
```diff
@@ -27,27 +27,31 @@ import (
...
 	return commonDataProvider{log, cfg, NewK8sDataProvider, NewAwsDataProvider}
 }
 
 func (c commonDataProvider) FetchCommonData(ctx context.Context) (CommonData, error) {
-	if c.cfg.Benchmark == "cis_eks" || c.cfg.Benchmark == "cis_k8s" {
-		return c.k8sDataProviderInit(c.log, c.cfg).FetchData(ctx)
-	}
+	var dataProviderInit EnvironmentDataProviderInit
+	switch c.cfg.Benchmark {
+	case "cis_eks", "cis_k8s":
```
Comment (uri-weisman) [inline dataprovider/common_data_provider.go:41]: nice

**Review by uri-weisman — APPROVED** [review]

**Pull request merged**